File: 0128 - Longest Consecutive Sequence/solution.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& A) {
        unordered_map<int, int> m;
        int res = 0;
        for(auto i : A){
            if(m[i]) continue;
            m[i] = m[i+m[i+1]] = m[i-m[i-1]] = m[i+1]+m[i-1]+1;
            res = max(res, m[i]);
        }
        return res;
    }
};

File: 0128 - Longest Consecutive Sequence/solution.py
class Solution:
    def longestConsecutive(self, A: List[int]) -> int:
        res, A = 0, set(A)
        for i in A:
            if i-1 not in A:
                r = i+1
                while r in A:
                    r += 1
                res = max(res, r-i)
        return res